addons/mail/controllers/thread.py

```diff
@@ -23,14 +23,37 @@ def mail_thread_data(self, thread_model, thread_id, request_list, **kwargs):
             ).get_result()
         return Store(thread, as_thread=True, request_list=request_list).get_result()
 
-    @http.route("/mail/thread/messages", methods=["POST"], type="json", auth="user")
-    def mail_thread_messages(self, thread_model, thread_id, search_term=None, before=None, after=None, around=None, limit=30):
-        domain = [
-            ("res_id", "=", int(thread_id)),
+    def _get_fetch_domain(self, thread_model, thread_id, **kwargs):
+        return [
+            ("res_id", "=", thread_id),
             ("model", "=", thread_model),
             ("message_type", "!=", "user_notification"),
         ]
-        res = request.env["mail.message"]._message_fetch(domain, search_term=search_term, before=before, after=after, around=around, limit=limit)
+
+    @http.route("/mail/thread/messages", methods=["POST"], type="json", auth="public")
+    def mail_thread_messages(
+        self,
+        thread_model,
+        thread_id,
+        search_term=None,
+        before=None,
+        after=None,
+        around=None,
+        limit=30,
```

Comment:
Maybe those should become an "options" dictionary forwarded to `_message_fetch` to limit the number of params?

Reply:
Done in #187192